### PR TITLE
[feat] pytorch lightning integration - training

### PR DESCRIPTION
**Pull request by ytsheng**
* pytorch lighting stub mostly involving training
* Tests for lightning trainer included

- [X] MVP 0. Training: Goal - Train a model from scratch and reach similar accuracy as using mmf_trainer
   - [X] Setup the training pipeline: done
   - [X] Training on the right device: done
   - [X] Clip gradients: done
   - [X] Optimizer: done
   - [X] LR scheduler (incl. warmup etc): done
   - [X] testcase: train visual_bert on vqa from scratch fo 10 iterations, compare the value: done
   - [X] tests... 

Note, the test is affected by: [Blocking] `end_train_batch_step` does not return train outputs properly: PR #[4369](https://github.com/PyTorchLightning/pytorch-lightning/pull/4369)

### CODE REVIEW

**Comment by ytsheng** [comment]
in favor of this: https://github.com/facebookresearch/mmf/pull/748
